libminifi/src/SchedulingAgent.cpp

```diff
@@ -125,6 +138,17 @@ bool SchedulingAgent::onTrigger(const std::shared_ptr<core::Processor> &processo
   return false;
 }
 
+void SchedulingAgent::watchDogFunc() {
+  std::lock_guard<std::mutex> lock(watchdog_mtx_);
+  auto now = std::chrono::system_clock::now();
```

Comment:
std::chrono::system_clock is not guaranteed to be monotonic, and it is indeed not in DST situations, manually setting the time or when using NTP. This can cause false positives and false negatives.
I think std::chrono::steady_clock should be used here.

Reply:
Done